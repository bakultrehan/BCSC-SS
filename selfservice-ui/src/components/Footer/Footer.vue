/** * Footer of app */

<template>
  <v-footer height="auto">
    <v-layout justify-center row wrap class="side-left-margin">
      <v-flex dark text-xs-center white--text xs12>
        <v-row justify="left" no-gutters>
          <a href="/" class="footer-link">Home</a>
          <a
            href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
            class="footer-link"
            target="_ss"
          >Disclaimer</a>
          <a
            href="https://www2.gov.bc.ca/gov/content/home/privacy"
            class="footer-link"
            target="_ss"
          >Privacy</a>
          <a
            href="https://www2.gov.bc.ca/gov/content/home/accessibility"
            class="footer-link"
            target="_ss"
          >Accessibility</a>
          <a
            href="https://www2.gov.bc.ca/gov/content/home/copyright"
            class="footer-link"
            target="_ss"
          >Copyright</a>
          <a href="/contact-us" class="footer-link">Contact Us</a>
        </v-row>
      </v-flex>
    </v-layout>
  </v-footer>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component
export default class Footer extends Vue {}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
footer {
  background-color: $BCgovBlue5 !important;
  border-top: 2px solid $BCgovGold5 !important;
  color: $BCgovWhite !important;
  padding: 4px 16px;
  z-index: 5;

  .footer-link {
    padding: 10px 16px;
    border-radius: unset;
    color: $BCgovWhite;
    text-decoration: none;
    font-size: 15px;
    @include sm {
      border-right: 1px solid $BCgovBlue3;
    }
  }
}
</style>
