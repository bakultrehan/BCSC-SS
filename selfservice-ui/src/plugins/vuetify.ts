import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
