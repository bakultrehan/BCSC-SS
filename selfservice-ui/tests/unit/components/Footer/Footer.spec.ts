import { shallowMount, createLocalVue } from '@vue/test-utils';
import Footer from '@/components/Footer/Footer.vue';
import vuetify from 'vuetify';

describe('Footer.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);

    wrapper = shallowMount(Footer, {
      localVue,
      mocks: { $t: jest.fn(() => {}) }, // tslint:disable-line
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
