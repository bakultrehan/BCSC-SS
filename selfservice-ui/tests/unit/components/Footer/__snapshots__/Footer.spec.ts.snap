// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue renders props when passed 1`] = `
<v-footer-stub
  height="auto"
  tag="div"
  tile="true"
>
  <v-layout-stub
    class="side-left-margin"
    justify-center=""
    row=""
    tag="div"
    wrap=""
  >
    <v-flex-stub
      dark=""
      tag="div"
      text-xs-center=""
      white--text=""
      xs12=""
    >
      <v-row-stub
        justify="left"
        nogutters="true"
        tag="div"
      >
        <a
          class="footer-link"
          href="/"
        >
          Home
        </a>
         
        <a
          class="footer-link"
          href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
          target="_ss"
        >
          Disclaimer
        </a>
         
        <a
          class="footer-link"
          href="https://www2.gov.bc.ca/gov/content/home/privacy"
          target="_ss"
        >
          Privacy
        </a>
         
        <a
          class="footer-link"
          href="https://www2.gov.bc.ca/gov/content/home/accessibility"
          target="_ss"
        >
          Accessibility
        </a>
         
        <a
          class="footer-link"
          href="https://www2.gov.bc.ca/gov/content/home/copyright"
          target="_ss"
        >
          Copyright
        </a>
         
        <a
          class="footer-link"
          href="/contact-us"
        >
          Contact Us
        </a>
      </v-row-stub>
    </v-flex-stub>
  </v-layout-stub>
</v-footer-stub>
`;
